terminus-ui/autocomplete/src/autocomplete.component.ts

```diff
@@ -1,67 +1,101 @@
 // tslint:disable: template-no-call-expression
+import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
 import {
+  AfterContentInit,
   AfterViewInit,
   ChangeDetectionStrategy,
   ChangeDetectorRef,
   Component,
+  ContentChildren,
   ElementRef,
   EventEmitter,
   Input,
   isDevMode,
   OnDestroy,
+  OnInit,
+  Optional,
   Output,
+  QueryList,
+  Self,
   ViewChild,
   ViewEncapsulation,
 } from '@angular/core';
 import {
   FormControl,
-  ValidationErrors,
+  NgControl,
 } from '@angular/forms';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+
 import {
-  MatAutocomplete,
-  MatAutocompleteSelectedEvent,
-  MatAutocompleteTrigger,
-} from '@angular/material/autocomplete';
+  BehaviorSubject,
+  of,
+  Subject,
+} from 'rxjs';
+
+import { CdkConnectedOverlay } from '@angular/cdk/overlay';
+import { MatChipList } from '@angular/material';
 import {
-  arrayContainsObject,
+  hasRequiredControl,
   isFunction,
+  TsDocumentService,
   untilComponentDestroyed,
 } from '@terminus/ngx-tools';
+import { coerceNumberProperty } from '@terminus/ngx-tools/coercion';
+import { TsFormFieldControl } from '@terminus/ui/form-field';
 import {
-  coerceArray,
-  coerceNumberProperty,
-} from '@terminus/ngx-tools/coercion';
+  TS_OPTION_PARENT_COMPONENT,
+  TsOptgroupComponent,
+  TsOptionComponent,
+} from '@terminus/ui/option';
 import { TS_SPACING } from '@terminus/ui/spacing';
 import { TsStyleThemeTypes } from '@terminus/ui/utilities';
-import { BehaviorSubject } from 'rxjs';
 import {
-  debounceTime,
-  distinctUntilChanged,
-  filter,
+  debounceTime, distinctUntilChanged, filter, switchMap,
 } from 'rxjs/operators';
+import {
+  TsAutocompletePanelComponent, TsAutocompletePanelSelectedEvent,
+} from './autocomplete-panel/autocomplete-panel.component';
+import { TsAutocompleteTriggerDirective } from './autocomplete-panel/autocomplete-trigger.directive';
 
 
+// Unique ID for each instance
+let nextUniqueId = 0;
+
 /**
  * Define a type for allowed {@link TsAutocompleteComponent} formatter function
  */
-// tslint:disable-next-line no-any
-export type TsAutocompleteFormatterFn<OptionType> = (value: OptionType) => string;
+export type TsAutocompleteFormatterFn = (value: string) => string;
 
 
 /**
  * Define a type for allowed {@link TsAutocompleteComponent} comparator function
  */
-export type TsAutocompleteComparatorFn<OptionType> = (value: OptionType) => string;
+export type TsAutocompleteComparatorFn = (value: string) => string;
+
 
 export class TsAutocompleteSelectedEvent extends MatAutocompleteSelectedEvent {}
 
-const DEFAULT_DEBOUNCE_MS = 200;
-const DEFAULT_MINIMUM_CHARACTERS = 2;
+const DEFAULT_MINIMUM_CHARACTER_COUNT = 2;
+const DEBOUNCE_DELAY = 200;
 
+/**
+ *  The formatter function type
+ */
+export type TsAutocompleteFormatFn = (v: string) => string;
+
+/**
+ * The event object that is emitted when the select value has changed
+ */
+export class TsAutocompleteChange<T = unknown> {
```

Comment:
I beleive we should be able to get rid of this generic now that we are only allowing `string`. (this line and line 93)